### PR TITLE
Commit changes for HW4

### PR DESCRIPTION
**Pull request by deepika-pdx**
Hello All,

Kindly review my changes for HW4.

### CODE REVIEW

**Comment by aashna15sethi** [comment]
reviewed

File: hw4/01-routing.js [hunk 1]
```diff
@@ -12,37 +14,60 @@ const port = process.env.PORT || 5001;
...
+  res.set({ "Content-Type": "text/plain" });
+  res.send("Your request to /redirect has been redirected to this page.");
+});
 
-// Add your code here
+app.get("/cache", (req, res) => {
+  res.status(200);
+  res.set("Cache-control", "public, max-age=86400");
+  res.set({ "Content-Type": "text/plain" });
+  res.send("This resource was cached");
```
Comment (sauravpdx) [inline hw4/01-routing.js:56]: It should be in HTML format not in plain text as per HW4

File: hw4/03-form.js [hunk 2]
```diff
@@ -1,16 +1,29 @@
...
+  formData = formData.concat("<b>Name: </b>").concat(`${req.body.username}`).concat(`<br>`);
+  formData = formData.concat("<b>Email: </b>").concat(`${req.body.email}`).concat(`<br>`);
+  const comments = req.body.comments !== "" ? req.body.comments : "n/a";
+  formData = formData.concat("<b>Comments: </b>").concat(`${comments}`).concat(`<br>`);
+  const newsletterCheck = req.body.newsletterCheck === "" ? "Yes, sign me up for the newsletter." : "No, thank you.";
+  formData = formData.concat("<b>Newsletter: </b>").concat(`${newsletterCheck}`);
+
+  res.set({ "Content-Type": "text/html" });
+  res.send(`${formData}`);
 });
```
Comment (sauravpdx) [inline hw4/03-form.js:27]: Just a suggestion: 
more cleaner would be to write all the HTML at once and keep inserting the variable in between. no need for concat

File: hw4/04-templates.js [hunk 3]
```diff
@@ -1,61 +1,137 @@
...
+    } else {
+      return countriesResponse.data;
+    }
+  } catch (error) {
+    console.error("There is some error with the get operation: ", error);
+  }
+}
+let countriesData = getCountriesData(url);
+
+app.get("/", (req, res) => {
```
Comment (sauravpdx) [inline hw4/04-templates.js:30]: Nice Implementation instead of writing the code again and again calling single function everytime

**Review by sauravpdx — APPROVED** [review]
good implementation deepika.. added few suggestion rest seems working

File: hw4/01-routing.js (hunk 1 above)
Comment (deepika-pdx) [reply hw4/01-routing.js:56]: Thanks. I just checked again. It is text/plain in hw4 for /cache.

**Pull request merged**